tests/dragonfly/tls_replication_test.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+import asyncio
+import redis
+from redis import asyncio as aioredis
+from . import DflyInstanceFactory, dfly_args
+import logging
+import time
+import subprocess
+from dragonfly.replication_test import check_all_replicas_finished
+
+BASE_PORT = 1111
+ADMIN_PORT = 1211
+
+"""
+Test replication with tls and non-tls options
+"""
+
+# 1. Number of master threads
+# 2. Number of threads for each replica
+replication_cases = [(8, 8)]
+
+tls_server_key_file_name = "df-key.pem"
+tls_server_cert_file_name = "df-cert.pem"
+
+def tls_args(dfly_path):
+    with_tls_args = {"tls": "",
+                    "tls_key_file": dfly_path + tls_server_key_file_name,
+                    "tls_cert_file": dfly_path + tls_server_cert_file_name,
+                    "no_tls_on_admin_port": "true"}
+    return with_tls_args
+
+def gen_tls_cert(dfly_path):
```

Comment:
consider making it a fixture (in conftest.py) similarly to other resources being consumed by tests.